### PR TITLE
Add golint workflow

### PR DESCRIPTION
**Pull request by roytman**
- Add a golint workflow. Currently, it checks only new code, eventually, we will validate all code. 
- In addition, this PR adds an option to configure goimpotrts with the separation of local imports.  in order to do it run `hack/tools/bin/fixgoimports.sh <fileName>. if filename is not provided, imports in te entire project will be corrected.
- Finally, it have an option to automatically verify code before commit. In order to do it, install `pre-commit` utility:
```
pip install pre-commit
pre-commit install
```

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -24,6 +24,8 @@ jobs:
       with:
         go-version: ${{ env.GO_VERSION }}
     - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
```
Comment (roee88) [inline .github/workflows/build.yml:28]: Why is this change required?

File: .github/workflows/golang-lint.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: golangci-lint
+        uses: reviewdog/action-golangci-lint@v2
+        with:
+          reviewdog_version: 'v0.13.1'
+          golangci_lint_version: 'v1.43.0'
+          golangci_lint_flags:  '--new-from-rev=HEAD~'
+          level: warning # GitHub Status Check won't become failure with this level.
```
Comment (roee88) [inline .github/workflows/golang-lint.yml:24]: I have been using the official action in other repos. It has good integration with GitHub. 
https://github.com/golangci/golangci-lint-action

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,148 @@
+linters-settings:
```
Comment (roee88) [inline .golangci.yml:1]: What is the original source of this configuration file? Asking because there are some stuff I don't understand (like why it has gci configuration)

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+repos:
```
Comment (roee88) [inline .pre-commit-config.yaml:1]: Need to add to development docs how to set it up. Also, for those of us who aren't fans of pre-commit hooks, I recommend that we make sure that the top IDEs can be setup to do all of these checks automatically and provide instructions on how to configure it so (not in this PR though)

File: hack/make-rules/tools.mk [hunk 5]
```diff
@@ -86,6 +86,13 @@ $(TOOLBIN)/misspell:
 	GOBIN=$(ABSTOOLBIN) go install github.com/client9/misspell/cmd/misspell@v0.3.4
 	$(call post-install-check)
 
+INSTALL_TOOLS += $(TOOLBIN)/goimports
+$(TOOLBIN)/goimports:
+	GOBIN=$(ABSTOOLBIN) go install golang.org/x/tools/cmd/goimports@v0.1.8
+	$(call post-install-check)
+	echo $(ABSTOOLBIN)/goimports -local fybrik.io/fybrik -w '$$'1 > $(ABSTOOLBIN)/fixgoimports.sh
+	chmod 755 $(ABSTOOLBIN)/fixgoimports.sh
+
```
Comment (roee88) [inline hack/make-rules/tools.mk:95]: I personally never liked the fact that tools.ml installs `dlv`. Now `goimports` falls into the same category in my opinion. The original idea was to have tools installed just for the project so they won't compete with a global install of the user. But in my opinion that doesn't hold for tools that I already use in the project because my IDE does (like `dlv` and `goimports`). Just a personal opinion though.

File: hack/make-rules/verify.mk [hunk 6]
```diff
@@ -38,43 +38,25 @@ CODE_MAINT += tidy
...
-.PHONY: lint-todo
-lint-todo: $(TOOLBIN)/golangci-lint
-	$(TOOLBIN)/golangci-lint run --enable=godox ${GOLINT_LINTERS} ./...
+.PHONY: lint-updated
+lint-updated: $(TOOLBIN)/golangci-lint
+	$(TOOLBIN)/golangci-lint run --new-from-rev=HEAD~
+
+.PHONY: lint-godox
+lint-godox: $(TOOLBIN)/golangci-lint
+	$(TOOLBIN)/golangci-lint run --enable=godox
```
Comment (roee88) [inline hack/make-rules/verify.mk:59]: Why is godox disabled in main config? Why does it need its own target? Also, godox is opiniated and we do use TODO comments in out code, like many other projects. You can change that of course :-)

**Review by roee88 — COMMENTED** [review]
1. It feels to me like there are too many lint related make targets and that can confuse people
2. revive can probably be removed from tools now

File: hack/make-rules/tools.mk (hunk 5 above)
Comment (roytman) [reply hack/make-rules/tools.mk:95]: indeed, but I wanted to allow to configure/fix imports separately. 
Plan to remove it later.

File: hack/make-rules/verify.mk (hunk 6 above)
Comment (roytman) [reply hack/make-rules/verify.mk:59]: it was one of my questions to you ;-)

File: .github/workflows/golang-lint.yml (hunk 2 above)
Comment (roee88) [reply .github/workflows/golang-lint.yml:24]: > but I want to add the reviewdog GitHub commiter

May I ask why?

Comment (roytman) [reply .github/workflows/golang-lint.yml:24]: sure, it simplifies to see why lint fails, but as I said it is a WIP, will see

Comment (roee88) [reply .github/workflows/golang-lint.yml:24]: What I meant is that the official action also shows this.

**Comment by roytman** [comment]
was done by #1340 and several other golint related PRs. Closing.
